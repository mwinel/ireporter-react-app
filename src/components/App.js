import React, { Component } from "react";
import { Provider } from 'react-redux';
import '../styles/App.css';
import store from '../redux/store';

class App extends Component {
  render () {
    return (
      <Provider store={store}>
        <div>
          <h1>Welcome to iReporter!</h1>
        </div>
      </Provider>
    );
  }
}

export default App;
