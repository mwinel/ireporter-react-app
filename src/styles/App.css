html, body {
  width: 100%;
  margin: 0;
}

body {
  font-family: Helvetica, sans-serif;
  font-size: 14px;
  color: #333333;
}

* {
  box-sizing: border-box;
}

.container {
  max-width: 980px;
  margin: auto;
}

.nav {
  background: #0c0c0c;
  text-align: right;
  height: 50px;
  line-height: 50px;
}

.brand {
  float: left;
  margin: 0 0 0 30px;
  background: #cc0000;
}

.brand:hover {
  opacity: 0.8;
  cursor: pointer;
}

.brand a {
  font-size: 15px;
  font-weight: 600;
  color: #ffffff;
  text-decoration: none;
  margin: 0 25px;
  line-height: 50px;
}

.menu {
  margin: 0 30px 0 0;
}

.menu a {
  clear: right;
  text-decoration: none;
  font-size: 15px;
  font-weight: 600;
  color: #a6a6a6;
  margin: 0 10px;
  line-height: 50px;
}

.menu a:hover {
  color: #ffffff;
}

label {
  margin: 0 25px 0 0;
  font-size: 26px;
  color: #a6a6a6;
  line-height: 50px;
  display: none;
  width: 30px;
  float: right;
}

.posted-title {
  margin-top: 45px;
}

#toggle {
  display: none;
}

#addUser {
  max-width: 380px;
  padding: 80px 25px;
  margin: 0 auto;
}

#userLogin {
  max-width: 380px;
  padding: 130px 25px;
  margin: 0 auto;
}

#dialogoverlay {
  display: none;
  opacity: .9;
  position: fixed;
  top: 0px;
  left: 0px;
  background: transparent;
  width: 100%;
  z-index: 10;
}

#dialogboxsuccess {
  display: none;
  position: fixed;
  border: 1px solid;
  border-radius: 4px;
  color: #155724;
  background-color: #d4edda;
  border-color: #c3e6cb;
  width: 400px;
  z-index: 10;
  font-size: 16px;
  scroll-behavior: auto;
}

#dialogboxsuccess > div {
  background: none;
}

#dialogboxsuccess > div #dialogboxbody-success {
  padding: 15px 20px 0px 20px;
}

#dialogboxsuccess > div #dialogboxfooter-success {
  padding: 15px 20px;
  float: right;
}

#dialogboxerror {
  display: none;
  position: fixed;
  border: 1px solid;
  border-radius: 4px;
  background-color: #f8d7da;
  border-color: #f5c6cb;
  width: 400px;
  z-index: 10;
  color: #721c24;
  font-size: 16px;
  scroll-behavior: auto;
}

#dialogboxerror > div {
  background: none;
}

#dialogboxerror > div #dialogboxbody-error {
  padding: 15px 20px 0px 20px;
}

#dialogboxerror > div #dialogboxfooter-error {
  padding: 15px 20px;
  float: right;
}

#alert-ok {
  color: #ffffff;
  background-color: #007bff;
  border-color: #007bff;
  border: 1px;
  border-radius: 3px;
  padding: 5px 10px;
}

.error-dialog {
  max-width: 380px;
  padding: 190px 25px;
  margin: 0 auto;
  text-align: center;
}

.error-heading {
  font-size: 80px;
  margin-top: 0px;
  margin-bottom: 0px;
}

.error-paragraph {
  font-size: 25px;
  margin-top: 0px;
  margin-bottom: 0px;
}

.form-footer {
  font-size: 12px;
}

input[type=text], [type=password] {
  width: 100%;
  padding: 12px;
  margin: 0 0 10px 0;
  font-size: 14px;
  border: 2px solid #a4a4a4;
  border-radius: 4px;
}

input[type=text]:focus, [type=password]:focus {
  outline: none;
  border: 2px solid #cc0000;
}

input[type=location] {
  width: 100%;
  padding: 12px;
  margin: 0 0 10px 0;
  font-size: 14px;
  border: 1px solid #a4a4a4;
  border-radius: 4px;
}

.btn-signup {
  padding: 12px 30px;
  font-size: 15px;
  background-color: #cc0000;
  color: #ffffff;
  font-weight: 600;
  border: none;
  border-radius: 4px;
  cursor: pointer;
}

.btn-login {
  padding: 12px 30px;
  font-size: 15px;
  background-color: #cc0000;
  color: #ffffff;
  font-weight: 600;
  border: none;
  border-radius: 4px;
  cursor: pointer;
}

.btn-signup:hover, .btn-login:hover {
  opacity: 0.7;
}

.btn-signup:focus, .btn-login:focus {
  outline: none;
}

.search-bar {
  padding: 50px 30px;
}

input[type=search] {
  border: none;
  background: #f6f4f6;
  /* display: inline; */
  width: 80%;
  padding: 16px;
  margin: 0 0 10px 0;
  font-size: 20px;
  border-radius: 4px;
}

input[type=search]:focus {
  outline: none;
  color: #333333;
}

.create-flag {
  flex: 30%;
  padding: 0px 30px;
}

.content {
  background: #f6f4f6;
  border-top: 1px solid #dbd9db;
  padding-top: 30px;
  padding-bottom: 30px;
}

.bc-link {
  background: #ffffff;
  padding: 8px 12px;
  border-radius: 4px;
  font-size: 12px;
  text-decoration: none;
  color: #333333;
}

.flag-content {
  margin: 0 30px;
}

.flag-title {
  margin-bottom: 30px;
}

.flag-card {
  background: #f6f4f6;
  padding-bottom: 40px;
  border-radius: 3px;
}

.flag-card img {
  border-top-left-radius: 3px;
  border-top-right-radius: 3px;
}

.card-title {
  margin: 15px 15px;
  font-size: 17px;
}

.card-desc {
  font-size: 13px;
  padding: 15px 15px;
  color: #333333;
}

.intervention {
  margin: 0 15px;
}

.btn-readmore {
  padding: 10px 15px;
  background: #cc0000;
  border: 2px solid #cc0000;
  border-radius: 4px;
  color: #ffffff;
  font-size: 14px;
  font-weight: 600;
  cursor: pointer;
  margin: 0px 15px 0px 15px;
  text-decoration: none;
}

.btn-readmore:hover {
  opacity: .7;
  color: #ffffff;
}

.btn-intervene:focus {
  outline: none;
}

.btn-create {
  padding: 16px 40px;
  background: #cc0000;
  border: 2px solid #cc0000;
  border-radius: 4px;
  color: #ffffff;
  font-size: 16px;
  font-weight: 600;
  cursor: pointer;
  margin-top: 5px;
  text-decoration: none;
}

.btn-create:hover {
  opacity: 0.8;
}

.btn-create:focus {
  outline: none;
}

.row:after {
  content: "";
  display: table;
  clear: both;
}

.section {
clear: both;
padding: 0px;
margin: 0px;
}

.col {
display: block;
float:left;
margin: 1% 1% 1% 0;
}
.col:first-child { 
  margin-left: 0; 
}

.group:before, .group:after { 
  content:""; display:table; 
}

.group:after { 
  clear:both;
}

.group { 
  zoom:1;
}

.span_3_of_3 { 
  width: 100%; 
}

.span_2_of_3 { 
  width: 66.13%; 
}

.span_1_of_3 { 
  width: 32.26%; 
}

.footer {
  margin: 30px 30px;
  border-top: 1px solid #dbd9db;
}

.copyright-text {
  font-size: 12px;
}

img {
  width: 100%;
}

#addIncident {
  max-width: 550px;
  padding: 0 25px;
  margin: 60px auto;
}

.upload {
  margin: 10px 0 10px 0;
  font-size: 13px;
}

textarea {
  width: 100%;
  height: 140px;
  padding: 14px;
  margin: 20px 0 10px 0;
  color: #495057;
  font-size: 14px;
  font-family: Helvetica;
  border: 1px solid #a4a4a4;
  border-radius: 4px;
}

textarea:focus {
  outline: none;
  border: 1px solid #cc0000;
}

input[type=title] {
  width: 100%;
  padding: 12px;
  margin: 0 0 10px 0;
  font-size: 14px;
  border: 1px solid #a4a4a4;
  border-radius: 4px;
}

input[type=title]:focus {
  outline: none;
  border: 1px solid #cc0000;
}

#incident_type {
  border: 1px solid #dbd9db;
  background: #f6f4f6;
  display: inline-block;
  padding: 12px;
  cursor: pointer;
  border-radius: 4px;
  width: 100%;
  margin-bottom: 10px;
  text-align: left;
  font-size: 13px;
}

.intervention {
clear: both;
padding: 0px;
/* margin: 0px; */
}

.column {
display: block;
float:left;
margin: 1% 0 1% 4%;
}

.column:first-child { 
  margin-left: 0; 
}

.group:before,
.group:after { 
  content:""; 
  display:table; 
}

.group:after { 
  clear:both;
}

.group { 
  zoom:1; /* For IE 6/7 */ 
}

.span_2_of_2 {
width: 34%;
}

.span_1_of_2 {
width: 60%;
}

.intervene h1, h3 {
  margin: 40px 0 20px 0;
}

.intervene p {
  font-size: 15px;
}

input[type=fname] {
  width: 100%;
  padding: 12px;
  font-size: 14px;
  border: 1px solid #a4a4a4;
  border-top-right-radius: 4px;
  border-top-left-radius: 4px;
}

input[type=lname] {
  width: 100%;
  padding: 12px;
  font-size: 14px;
  border-left: 1px solid #a4a4a4;
  border-right: 1px solid #a4a4a4;
  border-top: none;
  border-bottom: none;
}

input[type=email] {
  width: 100%;
  padding: 12px;
  font-size: 14px;
  border: 1px solid #a4a4a4;
  border-bottom-right-radius: 4px;
  border-bottom-left-radius: 4px;
}

input[type=fname]:focus, input[type=lname]:focus, input[type=email]:focus {
  outline: none;
}

.btn-intervention {
  padding: 12px 40px;
  margin: 20px 0;
  background: #cc0000;
  border: 2px solid #cc0000;
  border-radius: 4px;
  color: #ffffff;
  font-size: 14px;
  font-weight: 600;
  width: 100%;
  cursor: pointer;
}

.btn-intervention {
  opacity: 0.8;
  outline: none;
}

table {
  border-spacing: 0;
  width: 100%;
  border: 1px solid #ddd;
  background: #ffffff;
  border-radius: 4px;
  padding: 20px;
}

th, td {
  text-align: left;
  padding: 8px;
  font-size: 14px;
  font-family: helvetica;
}

.redflags {
  margin: 50px 0;
}

.redflags a {
  text-decoration: none;
}

.redflags a:hover {
  text-decoration: none;
}

tr:nth-child(even) {
  background-color: #f2f2f2
}

.table-bottom {
  font-weight: 600;
}

.btn-edit-redflag {
  background: #5563c1;
  border-radius: 3px;
  border: 1px solid #3f4eae;
  padding: 6px 10px;
  color: #ffffff;
  cursor: pointer;
}

.btn-delete-redflag {
  background: #cc0000;
  border-radius: 3px;
  border: 1px solid #cc0000;
  padding: 6px 10px;
  color: #ffffff;
  cursor: pointer;
}

.user-profile {
  margin-top: 40px;
}

@media only screen and (max-width: 500px) {
  .brand {
      margin: 0;
  }
  label {
      display: block;
      cursor: pointer;
  }
  label:hover {
      color: #ffffff;
  }
  .menu {
      text-align: center;
      width: 100%;
      display: none;
  }
  .menu a {
      display: block;
      border-bottom: 1px solid #a6a6a6;
      margin: 0;
      color: #ffffff;
      background: #DB584B;
  }
  #toggle:checked + .menu {
      display: block;
  }
  .form-group {
      width: 100%;
      margin: 0 auto;
  }
  .col {  
      margin: 1% 0 1% 0%; 
  }
  .column { 
  margin: 1% 0 1% 0%;
}
.span_3_of_3, .span_2_of_3, .span_1_of_3 { 
      width: 100%; 
  }
  .span_2_of_2, .span_1_of_2 { 
      width: 100%; 
  }
  .flag-card {
      margin-bottom: 20px;
  }
  input[type=search] {
      width: 100%;
  }
  .btn-create {
      width: 100%;
  }
}
