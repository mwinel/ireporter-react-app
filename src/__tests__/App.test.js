import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import App from '../components/App';

describe('App Component', () => {
  let wrapper = shallow(<App />)
  it('Should match snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
