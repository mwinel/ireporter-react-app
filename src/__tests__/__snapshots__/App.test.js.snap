// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component Should match snapshot 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <div>
    <h1>
      Welcome to iReporter!
    </h1>
  </div>
</Provider>
`;
